### ISSUE TITLE
crypto-bigint

### ISSUE THREAD

**Issue by tarcieri** [mikelodder7/unknown_order#1]
Hello!

I thought you might be interested in the `crypto-bigint` crate we've been working on as part of [RustCrypto](https://github.com/RustCrypto), which we are slowly beginning to use in our elliptic curve crates:

- https://crates.io/crates/crypto-bigint
- https://docs.rs/crypto-bigint

We're targeting Rust 1.51+ with it and aggressively trying to leverage both const generics and `const fn` as much as is possible on stable Rust.

We're targeting constant-time operation by default, with some secondary variable-time implementations explicitly called out as `*_vartime`.

I'm not sure we yet provide the functionality you need, or if the approach we're using (fixed-width, stack-allocated, `no_std` friendly big integers) actually maps well to this particular problem, but I thought I'd give it a mention.

If there's missing functionality you'd like to see implemented first, that's definitely something we'd like to know about:

https://github.com/RustCrypto/utils/issues/453

**Comment by mikelodder7** [comment]
Cool, thanks for the heads up. I'll check it out. The first I notice is I need the ability to represent negative numbers which often affects how computations happen in groups of unknown order.

**Comment by mikelodder7** [comment]
Crypto-bigint will be a nice addition for going to no-std environments and constant-time operations. Let me know how I can help.

**Comment by davfra** [comment]
What is the current status on this? 
Can you guesstimate the effort and/or steps needed to include [crypto-bigint](https://crates.io/crates/crypto-bigint) in 
unknown_order or comment on the obstacles?

**Comment by mikelodder7** [comment]
There’s some methods missing from crypto-bigint I’m working to add then it will be quick

**Comment by davfra** [comment]
Cool! Is there anything specific I could help with?

**Comment by mikelodder7** [comment]
You stub it in for now and implement what can be done

**Comment by donkey-donkey** [comment]
hi, wondering if this will help with some of the no_std. trying to use this unknown_order presently and getting a std error from num-traits

**Comment by mikelodder7** [comment]
I'm actually close to having crypto-bigint almost integrated. Expect it within a week or two

**Comment by donkey-donkey** [comment]
rad. super stoked on your timing. thanks

**Comment by donkey-donkey** [comment]
rad. super stoked on your timing. thanks

**Comment by donkey-donkey** [comment]
checking in on this to see if there is anything that we can test? thanks

**Comment by mikelodder7** [comment]
Just finishing some testing. I'll push what I have. Also doing some code cleanup

**Comment by donkey-donkey** [comment]
awesome stoked to try it out. 
thanks.

**Comment by mikelodder7** [comment]
Sorry I'm slow on this. Been traveling a ton and had zero time. Just a minor issue with serialization then I think I've got it

**Comment by mikelodder7** [comment]
Published an RC candidate `unknown_order = "0.7.0-rc0".  The one issue I'm having is when you try to serialize to json or toml or anything human readable I get

**serializer is human readable, which requires the `alloc` crate feature**.

Have you ever seen this @tarcieri with crypto-bigint?

**Comment by donkey-donkey** [comment]
hi. any updates. i noticed  the repo  is now with hyperledger. are you still working on it? thanks much.

**Comment by mikelodder7** [comment]
Yes I've just been really busy. Take a look and see if you can figure out why the serialize bug happens. That's the only hinderance right now.

**Comment by mikelodder7** [comment]
Addressed in #9
